src/bin/pgcopydb/catalog.c

```diff
@@ -7700,6 +7707,176 @@ catalog_count_summary_done_fetch(SQLiteQuery *query)
 }
 
 
+/*
+ * catalog_add_timeline_history is a wrapper for
+ * catalog_add_timeline_history_internal so that callers can be agnostic of the
+ * TimelineHistoryContext.
```

Comment:
I'm not sure I see the point about that yet. Will get back to it in another round of review I suppose.

Reply:
After introducing a new module, this dirty workaround is not really necessary. I merged the two functions into one.